server/services/phase.ts

```diff
@@ -111,11 +111,22 @@ export const getAllSitePhases = async (siteId: number): Promise<sitePhase[]> =>
   }));
 };
 
-export const getAllPhases = async () => {
+export const getAllPhases = async (query = null) => {
```

Comment:
Document what this new query parameter does with JSDoc, and add a type. Neither is immediately clear at the moment, and ideally we should be striving for as much strictly defined typing as possible.

The name is also a potential source of confusion - I'd expect a parameter called `query` to be something like a SQL query in this region of the codebase, but here it looks like, presumably, query parameters from a request? I haven't had a chance to check in-IDE or look in greater depth, but that's what it looks like from here.

Considering all you're doing with this property is reading a bool from it, I'd highly suggest just replacing `query` with an `includeAllocations` parameter, and adjusting your call to `getAllPhases` to pass it that specific property of the query object.

Reply:
Also - I created a `phase` type as it was a different response than `sitePhase`, which exposed the inconsistencies of the response in regards to casing (snake vs camel). I'm making it OOS for this ticket as the API was already implemented and there is still lots of testing to complete before EOS, but it's a Good Candidate for a tech debt ticket to update the casing, and decide on preferred formatting for APIs moving forward. (again, if there is time after testing I can update)

Reply:
Good point in regards to casing inconsistency, let's have a discussion about how we want to handle that later, and make a ticket for it. I like the new changes you made here.